receiptui/src/test/java/com/hyperwallet/android/ui/receipt/viewmodel/ListPrepaidCardReceiptViewModelTest.java

```diff
@@ -15,42 +19,53 @@
 @RunWith(RobolectricTestRunner.class)
 public class ListPrepaidCardReceiptViewModelTest {
 
-    private ReceiptViewModel mReceiptViewModelToTest;
+    private ReceiptViewModel mReceiptViewModel;
     private ListPrepaidCardReceiptViewModel.ListPrepaidCardReceiptViewModelFactory
-            mListPrepaidCardReceiptViewModelFactoryToTest;
+            mListPrepaidCardReceiptViewModelFactory;
+    private PrepaidCardReceiptRepository mPrepaidCardReceiptRepository;
 
     @Before
     public void initializedViewModel() {
-        PrepaidCardReceiptRepository prepaidCardReceiptRepository = new PrepaidCardReceiptRepositoryImpl(
-                "trm-ppc-token");
-        mListPrepaidCardReceiptViewModelFactoryToTest =
+        mPrepaidCardReceiptRepository = spy(new PrepaidCardReceiptRepositoryImpl("trm-ppc-token"));
+        mListPrepaidCardReceiptViewModelFactory =
                 new ListPrepaidCardReceiptViewModel.ListPrepaidCardReceiptViewModelFactory(
-                        prepaidCardReceiptRepository);
-        mReceiptViewModelToTest = mListPrepaidCardReceiptViewModelFactoryToTest.create(ReceiptViewModel.class);
+                        mPrepaidCardReceiptRepository);
+        mReceiptViewModel = mListPrepaidCardReceiptViewModelFactory.create(ReceiptViewModel.class);
     }
 
     @Test
     public void testIsLoadingData_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.isLoadingData(), is(notNullValue()));
+        assertThat(mReceiptViewModel.isLoadingData(), is(notNullValue()));
     }
 
     @Test
     public void testGetReceiptErrors_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getReceiptErrors(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getReceiptErrors(), is(notNullValue()));
     }
 
     @Test
     public void testGetReceiptList_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getReceiptList(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getReceiptList(), is(notNullValue()));
     }
 
     @Test
     public void testGetDetailNavigation_returnsLiveData() {
-        assertThat(mReceiptViewModelToTest.getDetailNavigation(), is(notNullValue()));
+        assertThat(mReceiptViewModel.getDetailNavigation(), is(notNullValue()));
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testListPrepaidCardReceiptViewModelFactory_throwsExceptionOnInvalidClassArgument() {
-        mListPrepaidCardReceiptViewModelFactoryToTest.create(ReceiptDetailViewModel.class);
+        mListPrepaidCardReceiptViewModelFactory.create(ReceiptDetailViewModel.class);
+    }
+
+    @Test
+    public void testListPrepaidCardReceiptViewModel() {
+        verify(mPrepaidCardReceiptRepository, never()).loadPrepaidCardReceipts();
+    }
+
+    @Test
+    public void testInit() {
+        mReceiptViewModel.init();
+        verify(mPrepaidCardReceiptRepository, times(1)).loadPrepaidCardReceipts();
```

Comment:
the default is already execute once so no need to explicitly specify

Reply:
@manikandan-altmtrk , @peter-joseph is correct, the method init will be called every time a configuration change occurs. Please add a boolean variable in the view model classes, and only call the repository method once

```suggestion
        public void init() {
        if (!inInitialized) {
            inInitialized = true;
            if (mSourceToken == null) {
                loadTransferSource();
            } else {
                loadTransferDestination(mSourceToken);
            }
        }
    }
```

Reply:
Done.